website/static/js/osfHelpers.js

```diff
@@ -499,6 +500,126 @@ ko.bindingHandlers.listing = {
 };
 
 
+
+/**
+ * Determines if an object is a function
+  * @param {object} obj
+ */
+var isFunction = function (obj) {
+    return (typeof(obj) === 'function' && typeof(obj.call) === 'function' && typeof(obj.apply) === 'function');
+};
+
+/**
+*  returns a random name from this list to use as a confirmation string
+*/
+var _confirmationString = function() {
+    // TODO: Generate a random string here instead of using pre-set values
+    //       per Jeff, use ~10 characters
+    var scientists = [
+        'Anning',
+        'Banneker',
+        'Cannon',
+        'Carver',
+        'Chappelle',
+        'Curie',
+        'Divine',
+        'Emeagwali',
+        'Fahlberg',
+        'Forssmann',
+        'Franklin',
+        'Herschel',
+        'Hodgkin',
+        'Hopper',
+        'Horowitz',
+        'Jemison',
+        'Julian',
+        'Kovalevsky',
+        'Lamarr',
+        'Lavoisier',
+        'Lovelace',
+        'Massie',
+        'McClintock',
+        'Meitner',
+        'Mitchell',
+        'Morgan',
+        'Odum',
+        'Pasteur',
+        'Pauling',
+        'Payne',
+        'Pearce',
+        'Pollack',
+        'Rillieux',
+        'Sanger',
+        'Somerville',
+        'Tesla',
+        'Tyson',
+        'Turing'
+    ];
+
+    return scientists[Math.floor(Math.random() * scientists.length)];
+};
+
+/**
+  * Confirm a dangerous action by requiring the user to enter specific text
+  *
+  * This is an abstraction over bootbox, and passes most options through to
+  * bootbox.dailog(). The exception to this is `callback`, which is called only
+  * if the user correctly confirms the action.
+  *
+  * @param  {Object} options
+  */
+var confirmDangerousAction = function (options) {
+    // TODO: Refactor this to be more interactive - use a ten-key-like interface
+    //       and display one character at a time for the user to enter. Once
+    //       they enter that character, display another. This will require more
+    //       sustained attention and will prevent the user from copy/pasting a
+    //       random string.
+
+    if (typeof(options) === 'undefined') {
+        options = {};
+    }
+
+    // set default values
+    if (!options.title) { options.title = 'Confirm Action'; }
```

Comment:
Use the standard pattern of defining a default options object and extending it with the passed in options.

Along these lines:

```js
var defaults = {
    title: 'Confirm action',
    confirmText: _confirmationString,
    buttons: //...
}
var opts = $.extend({}, defaults, options);

var bootboxOptions = $.extend({}, opts, {
    message: [
        '<p>Type the following to continue: <strong>',
        typeof opts.confirmText === 'function' ? opts.confirmText() : opts.confirmText,
        '</strong></p>',
        // ...
    ].join(''),
    callback: ...,
})

bootbox.dialog(bootboxOptions)
```